src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,196 @@
+package duke.storage;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.exception.DataFileChangedException;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+/**
+ * Handles loading and saving tasks from the data file.
+ */
+public class Storage {
+
+    /** Path to data file. */
+    private final String filePath;
+
+    /**
+     * Constructor for Storage class.
+     *
+     * @param filePath path to data file.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Saves and writes the current tasks to the data file.
+     * If the file does not exist, it will create a new file.
+     *
+     * @param taskList contains all current tasks.
+     * @throws IOException if the named file exists but is a directory
+     * rather than a regular file, does not exist but cannot be created,
```

Comment:
Should we indent continuation lines four (or more) spaces from the position of the `@`? I noticed the same issue in several other places too.